packages/wdio-browserstack-service/src/accessibility-handler.ts

```diff
@@ -97,31 +104,61 @@ class _AccessibilityHandler {
         (this._browser as WebdriverIO.Browser).getAccessibilityResults = async () => {
             return await getA11yResults((this._browser as WebdriverIO.Browser), isBrowserstackSession(this._browser), this._accessibility)
         }
-    }
 
-    async beforeTest (suiteTitle: string | undefined, test: Frameworks.Test) {
-        if (
-            this._framework !== 'mocha' ||
-            !this.shouldRunTestHooks(this._browser, this._accessibility)
-        ) {
-            return
+        (this._browser as WebdriverIO.Browser).performScan = async () => {
+            return await performA11yScan((this._browser as WebdriverIO.Browser), isBrowserstackSession(this._browser), this._accessibility)
         }
 
-        const shouldScanTest = shouldScanTestForAccessibility(suiteTitle, test.title, this._accessibilityOptions)
-        const testIdentifier = this.getIdentifier(test)
-        const isPageOpened = await this.checkIfPageOpened(this._browser, testIdentifier, shouldScanTest)
-
-        if (!isPageOpened) {
-            return
+        if (this._accessibility) {
```

Comment:
Let's return early. Can we do this for the following `if` statements too?
```suggestion
        if (!this._accessibility) {
            return
        }
```